libraries/libfc/include/fc/time.hpp

```diff
@@ -49,6 +49,16 @@ namespace fc {
         std::string to_iso_string()const;
         static time_point from_iso_string( const std::string& s );
 
+        // protect against overflow
+        constexpr time_point& safe_add( const microseconds& m ) {
+           if (m.count() > 0 && elapsed > fc::microseconds::maximum() - m) {
+              elapsed = microseconds::maximum();
+           } else { // does not guard against underflow
```

Comment:
I could but as the comment says it does not guard against underflow. My thought was that in nominal use m would never be negative so don't pay the cost of checking. Happy to add a true check if you think we should have it. Probably should since caller may expect it.

Reply:
OK, we don't want to pay the cost of check, if it would never be negative. Maybe the name should be changed bounded_add or something. But you don't need to change.